### PR TITLE
Release/1.0.0

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
Initial Commit

### CODE REVIEW

File: widget/edit.controller.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        $scope.config = config;
+        $scope.config.eventName = $scope.config.eventName ? $scope.config.eventName : "";
+        $scope.config.broadcastEvent = $scope.config.broadcastEvent ? $scope.config.broadcastEvent : false;
+        $scope.loadAttributesForCustomModule = loadAttributesForCustomModule;
+        $scope.jsonObjModuleList = [];
+        $scope.toggleArrow = toggleArrow;
+
+        init();
+        function init() {
+            appModulesService.load(true).then(function (modules) {
```
Comment (cs-bk) [inline widget/edit.controller.js:27]: @cs-yash-patodkar - this code is being in all the widgets

File: widget/edit.controller.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                        //Check If JSON field is present in the module
+                        if (moduleMetaData.attributes[fieldIndex].type === "object") {
+                            $scope.jsonObjModuleList.push(module);
+                            break;
+                        }
+                    }
+                })
+            })
+        }
+        if ($scope.config.customModule) {
```
Comment (cs-bk) [inline widget/edit.controller.js:42]: @cs-yash-patodkar - why is if needed

File: widget/edit.html [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                data-ng-class="{ 'has-error': editRecordSummaryCard.customModule.$invalid && editRecordSummaryCard.customModule.$touched }">
+                <label for="customModule" class="control-label">Data Source
+                    <span class="text-danger">*</span>
+                </label>
+                <span data-uib-tooltip="Select the Module which has JSON data" data-tooltip-append-to-body="true"><i
+                        class="icon icon-information font-Size-13"></i></span>
+                <select name="customModule" id="customModule" class="form-control"
+                    data-ng-options="module.type as module.name for module in jsonObjModuleList"
+                    data-ng-model="config.customModule"
+                    data-ng-change="loadAttributesForCustomModule(config.customModule)" required>
```
Comment (cs-bk) [inline widget/edit.html]: @cs-yash-patodkar - loadAttributesForCustomModule(config.customModule), but in controller the function does not take any parameter

File: widget/view.controller.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      var filters = {
+        query: _config.query
+      };
+      var pagedTotalData = new PagedCollection(_config.customModule, null, null);
+      pagedTotalData.loadByPost(filters).then(function () {
+        if (pagedTotalData.fieldRows.length === 0) {
+          $scope.filterValidation = true;
+          return;
+        }
+        for (let i = 0; i < pagedTotalData.fieldRows.length; i++) {
```
Comment (cs-bk) [inline widget/view.controller.js:49]: @cs-yash-patodkar - lets go through this logic over call

File: widget/view.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        </div>
+        <div class="padding-right-0 padding-left-0"
+            data-ng-class="(page === 'dashboard' || page === 'reporting') ? 'widget-dashboard-actions-width' : 'widget-actions-width'">
+            <span class="fa btn widget-action-icon btn-sm pull-right" data-ng-click="collapsed = !collapsed"
+                data-ng-class="{'fa-caret-up': !collapsed, 'fa-caret-down': collapsed}"></span>
+        </div>
+    </div>
+    <div data-ng-hide="collapsed">
+        <div id="globalCard-{{$index}}" class="global-card-dark display-inline-block" ng-repeat="item in widgetData"
+            data-ng-click="runCommand(widgetData[$index]['@id'], $index)">
```
Comment (cs-bk) [inline widget/view.html]: @cs-yash-patodkar - should be clickable only if broadcast event is selected

File: widget/edit.controller.js (hunk 2 above)
Comment (cs-yash-patodkar) [reply widget/edit.controller.js:42]: To get the fields of module, not storing the fields in config.

File: widget/info.json
Comment (cs-yash-patodkar) [inline widget/info.json:1]: Add reports

**Review by cs-bk — APPROVED** [review]
LGTM

**Review by cs-suyog-jinturkar — APPROVED** [review]
BK already approved this PR, hence marking it as approved here as well.

**Pull request merged**